erpnext/education/doctype/student_leave_application/student_leave_application.py

```diff
@@ -4,7 +4,28 @@
 
 from __future__ import unicode_literals
 import frappe
+from frappe import _
 from frappe.model.document import Document
 
 class StudentLeaveApplication(Document):
-	pass
+	def validate(self):
+		self.validate_duplicate()
+
+	def validate_duplicate(self):
+		data = frappe.db.sql(""" select name from `tabStudent Leave Application`
+			where
+				((%(from_date)s > from_date and %(from_date)s < to_date) or
+				(%(to_date)s > from_date and %(to_date)s < to_date) or
+				(%(from_date)s <= from_date and %(to_date)s >= to_date)) and
+				name != %(name)s and student = %(student)s and docstatus < 2
+		""", {
+			'from_date': self.from_date,
+			'to_date': self.to_date,
+			'student': self.student,
+			'name': self.name
+		}, as_dict=1)
+
+		if data:
+			link = "<a href='#Form/Student Leave Application/{0}'>{1}</a>".format(data[0].name, data[0].name)
+			frappe.throw(_("Leave application {0} is already exists for the same period against the student {1}")
```

Comment:
This is minor, but it should read `{0} already exists` and `against student {1}`.